### PR TITLE
Performance improvements for `build_histogram` and `adjust_histogram`

### PR DESCRIPTION
**Pull request by timholy**
Closes https://github.com/JuliaImages/Images.jl/pull/855

### CODE REVIEW

File: src/build_histogram.jl [hunk 1]
```diff
@@ -127,23 +146,50 @@ edges, counts  = build_histogram(r, 256, minval = 0, maxval = 1)
...
+    build_histogram(mappedarray(Gray{N0f8}, img))
+
+function build_histogram(img::AbstractArray{T}, nbins::Integer;
+                         minval::Union{Real,AbstractGray}=minfinite(img),
+                         maxval::Union{Real,AbstractGray}=maxfinite(img))  where T<:Union{N0f8, AbstractGray{N0f8}}
+    edgesraw, countsraw = build_histogram(img)
+    return rebin(edgesraw, countsraw, nbins, minval, maxval)
+end
+
+# build_histogram(img, edges)
```
Comment (zygmuntszpak) [inline src/build_histogram.jl:181]: Did you intend to leave this comment here?

**Comment by zygmuntszpak** [comment]
Thank you Tim. Your implementation is always an education :D.

**Review by zygmuntszpak — APPROVED** [review]

File: src/build_histogram.jl (hunk 1 above)
Comment (timholy) [reply src/build_histogram.jl:181]: Yeah, kind of a "section header" (the ones above use `nbins`, the ones below `edges`). But I can get rid of it if you don't like it.

Comment (zygmuntszpak) [reply src/build_histogram.jl:181]: It is fine. I assumed you meant it as a section header, but just wanted to confirm.

**Pull request merged**